addons/sale/static/src/js/product_list/product_list.xml

```diff
@@ -12,8 +12,15 @@
             <thead t-if="!this.props.areProductsOptional">
                 <tr>
                     <th class="px-0 border-bottom-0" colspan="2">Product</th>
-                    <th class="px-0 text-center border-bottom-0">Quantity</th>
-                    <th class="px-0 text-end border-bottom-0">Price</th>
+                    <th class="px-0 text-center border-bottom-0" t-if="env.showQuantity">
```

Comment:
```suggestion
                    <th t-if="env.showQuantity" class="px-0 text-center border-bottom-0">
```
nit: `t-if` should always be the first attribute, as it decides whether the whole node will be rendered.